client/profile-wizard/steps/business-details/flows/selective-bundle/selective-extensions-bundle/index.js

```diff
@@ -0,0 +1,341 @@
+							link: (
+								<Link
+									href="https://wordpress.com/tos/"
+									target="_blank"
+									type="external"
+								/>
+							),
+						},
+					} ) }
+				</Text>
+			) }
+		</div>
+	);
+};
+
+export const SelectiveExtensionsBundle = ( {
+	isInstallingActivating,
+	onSubmit,
+} ) => {
+	const [ showExtensions, setShowExtensions ] = useState( false );
+	const [ values, setValues ] = useState( initialValues );
+
+	return (
+		<div className="woocommerce-profile-wizard__business-details__free-features">
+			<Card>
+				<div className="woocommerce-profile-wizard__business-details__free-features__illustration">
+					<AppIllustration />
+				</div>
+				<div className="woocommerce-admin__business-details__selective-extensions-bundle">
+					<div className="woocommerce-admin__business-details__selective-extensions-bundle__extension">
+						<CheckboxControl
+							id="woocommerce-business-extensions__checkbox"
+							checked={ values.install_extensions }
+							onChange={ ( checked ) => {
+								setValues(
+									setAllPropsToValue( values, checked )
+								);
+							} }
+						/>
+						<p className="woocommerce-admin__business-details__selective-extensions-bundle__description">
+							{ __(
+								'Add recommended business features to my site'
+							) }
+						</p>
+						<Icon
+							className="woocommerce-admin__business-details__selective-extensions-bundle__expand"
+							icon={ showExtensions ? chevronUp : chevronDown }
+							onClick={ () => {
+								const shouldShowExtensions = ! showExtensions;
+								setShowExtensions( shouldShowExtensions );
+
+								if ( shouldShowExtensions ) {
+									// only record the accordion click when the accordion is opened.
+									recordEvent(
+										'storeprofiler_store_business_features_accordion_click'
+									);
+								}
+							} }
+						/>
+					</div>
+					{ showExtensions &&
+						installableExtensions.map( ( { plugins, title } ) => (
+							<div key={ title }>
+								<div className="woocommerce-admin__business-details__selective-extensions-bundle__category">
+									{ title }
+								</div>
+								{ plugins.map( ( { description, slug } ) => (
+									<div
+										key={ slug }
+										className="woocommerce-admin__business-details__selective-extensions-bundle__extension"
+									>
+										<CheckboxControl
+											id="woocommerce-business-extensions__checkbox"
+											checked={ values[ slug ] }
+											onChange={ ( checked ) => {
+												const newState = {
+													...values,
+													[ slug ]: checked,
+												};
+
+												const allExtensionsDisabled =
+													Object.entries(
+														newState
+													).filter(
+														( [ , val ] ) => val
+													).length === 1 &&
+													newState.install_extensions;
+
+												if ( allExtensionsDisabled ) {
+													// If all the extensions are disabled then disable the "Install Extensions" checkbox too
```

Comment:
I clearly need a wider monitor...

Reply:
I blame all the spacing in our code standard :P

Reply:
@becdetat Seriously though, this is something that I always go back and forth on. its easier for the reader for the code to be in place on the control to figure out "whats going on", than say storing the callback somewhere else and forcing them to navigate there to figure out whats going on.

Genuinely our coding standard harms this approach with tabs though... If you have a suggestion of what you would prefer its easy enough for me to make the change.

Reply:
Every time I have a `.map` I try to break the contents down into a smaller function. So you would have:

```
installableExtensions.map( { plugins, title } => getInstallableExtension( plugins, title ) )
```

and then in `getInstallableExtension` you've got a `plugins.map` so that would be broken down into another smaller method. Yes you need to navigate to find the code for the mapping but if you can name the map functions clearly that isn't usually a big deal as you've got small functions that don't do as much.

Reply:
Thats fair, I can move this into a callback or a separate component.

Reply:
One thing I would say is that this doesn't look crazy in VSCode or require a wide screen. Its just that github renders very wide tabs. However I think it was an improvement to move this rendering into a separate component. I moved the onChange callback out as well, so hopefully thats more readable: bd57378